src/ripple/overlay/impl/PeerImp.cpp

```diff
@@ -1071,9 +1073,10 @@ PeerImp::onMessage (std::shared_ptr <protocol::TMTransaction> const& m)
             std::weak_ptr<PeerImp> weak = shared_from_this();
             app_.getJobQueue ().addJob (
                 jtTRANSACTION, "recvTransaction->checkTransaction",
-                [weak, flags, stx] (Job&) {
+                [weak, flags, checkSignature, stx] (Job&) {
```

Comment:
@ximinez It might look a bit ugly but its definitely preferred because 1. It reduces the scope of `weak` to the lambda and 2. It avoids a needless move or copy